### DISCUSSION TITLE
Improve PR with automated tools

### DISCUSSION THREAD

**Discussion by farodin91** [JanusGraph/janusgraph#2312] [General]
Tools are we using or had used:
- codecov - https://codecov.io/gh/JanusGraph/janusgraph (mainly coverage)
- coverity - https://scan.coverity.com/projects/janusgraph-janusgraph (security focused, only manually)
- codacy - https://app.codacy.com/manual/JanusGraph/janusgraph/dashboard?branch=master (code quality checks) 

New Tools
- lgtm - https://lgtm.com/projects/g/JanusGraph/janusgraph/?mode=list (small list of quality checks)
- codeql (internal security scan, see https://github.com/JanusGraph/janusgraph/security/code-scanning)
---
Did I forgot anything? What do we want to do? 

My Idea would be to activate codecov and lgtm in PRs.

**Comment by li-boxuan** [comment]
IIRC codacy was enabled before. I found it pretty useful. Did someone disable it?

lgtm looks similar to codacy. It seems codacy is more popular than lgtm.

I definitely vote for enabling codecov in PRs. I often visit the website and check coverage diff manually. Would be nice to see codecov diff in PR directly.

**Reply by farodin91** [reply]
For me codacy, it was pretty annoying and also we had a lot of false positives.

**Reply by farodin91** [reply]
codacy can be converted into an app than it should work much bettter: https://docs.codacy.com/faq/repositories/how-do-i-migrate-a-legacy-repository-to-a-synced-organization/

**Reply by farodin91** [reply]
@mbrukman Do you have admin rights in janusraph and codacy? Could you convert our codacy installation? 

I think we(@porunov and I) are able to migrate codacy.

**Reply by li-boxuan** [reply]
Do we also consider checkstyle?

**Reply by farodin91** [reply]
Would be a good but they should be enforce also by standard editors such as VScode or Intellij.

**Reply by farodin91** [reply]
Codacy is back online.

**Reply by li-boxuan** [reply]
I just loosened the Codacy quality check requirement. Now it requires no major issue (previously it requires no minor issue).
